addons/base_vat/views/res_partner_views.xml

```diff
@@ -5,8 +5,20 @@
             <field name="model">res.partner</field>
             <field name="inherit_id" ref="base.view_partner_form"/>
             <field name="arch" type="xml">
-                <xpath expr="//field[@name='vat']" position="attributes">
-                    <attribute name="string">VAT</attribute>
+                <xpath expr="//field[@name='vat']" position="replace">
+                </xpath>
+
+                <xpath expr="//label[@for='street']" position="before">
+                    <field name="base_vat_check_status" invisible="1"/>
+                    <field name="vat" string="VAT" style="color:Red;" attrs="{'invisible': ['|', ('base_vat_check_status', '!=', 'wrong'), ('vat','=',False)]}"/>
+                    <field name="vat" string="VAT" style="color:Green;" attrs="{'invisible': [('base_vat_check_status', '!=', 'verified')]}"/>
+                    <field name="vat" string="VAT" attrs="{'invisible': [('vat','!=',False)]}"/>
+
+                    <label string="VAT" style="font-weight: bold;" attrs="{'invisible': ['|', ('base_vat_check_status', '!=', 'unknown'), ('vat','=',False)]}"/>
+                    <div class="o_row" attrs="{'invisible': ['|', ('base_vat_check_status', '!=', 'unknown'), ('vat','=',False)]}">
+                        <field name="vat" string="VAT" style="color:Orange;"/>
+                        <button id="base_vat_recheck_vies" string="Recheck" type="object" name="compute_base_vat_check_status" class="oe_read_only btn-default"/>
+                    </div>
```

Comment:
the colouring of the field is crappy, hence there's currently no other way of doing it in a better way.

As discussed with @ged-odoo , you need to adapt the web client to allow decoration-xxxx attributes to be set on any node of the form view (currently only possible for tree views)

So the definition of the VAT field would simply become:
```xml
<field name="vat" string="VAT" decoration-danger="base_vat_check_status=='wrong'" decoration-warning="base_vat_check_status=='unknown'" decoration-success="base_vat_check_status=='verified'"/>
```